addons/mail/static/src/models/message.js

```diff
@@ -345,10 +345,21 @@ registerModel({
                 return !this.messaging.isCurrentUserGuest && !this.isTemporary && !this.isTransient;
             },
         }),
+        date: attr(),
         /**
          * Determines the date of the message as a moment object.
          */
-        date: attr(),
+        momentDate: attr({
+            compute() {
+                if (!this.date) {
+                    return;
+                }
+                if (!moment.isMoment(this.date)) {
+                    return moment(str_to_datetime(this.date));
+                }
+                return this.date;
```

Comment:
Don't think we want to return `this.date` here. Better also `clear()`, so that we just deal with `moment` or a cleared value

Reply:
No, we need to return `this.date` because it could already be a moment object in some cases.

Reply:
The condition your are mentioning is testing that this.date is NOT a moment object already. Not that it is one.

Reply:
Ok my bad, I have a hard time seeing the `!` when there's a preceding open bracket...
Looks good then!